src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,14 @@ protected static function flatten_tree( $tree, $prefix = '', $token = '--' ) {
 	 * @return array  Returns the modified $declarations.
 	 */
 	protected static function compute_style_properties( $styles, $settings = array(), $properties = null, $theme_json = null, $selector = null, $use_root_padding = null ) {
+		$args      = func_get_args();
+		$cache_key = 'compute_style_properties_' . md5( wp_json_encode( $args ) );
+		$cache     = wp_cache_get( $cache_key, 'wp-styles' );
```

Comment:
Question: If we introduce a development mode check, we might end up in a loop of discussions similar to what we had in [#59719](https://core.trac.wordpress.org/ticket/59719). Would it be advisable to prioritize and address that ticket first to determine the best way forward, making it easier to adopt the solution in other places? What are your thoughts on this?

Reply:
Good question—thanks for asking, @mukeshpanchal27. I don't think we need that discussion to block forward progress on this ticket since it's not proposing that we make these caches persistent via transients. In this case, adding a development mode check just ensures that anyone who is developing while using an object cache would still get fresh results.

Reply:
@mukeshpanchal27 @joemcgill, thanks for these inputs.

I just pushed an update to bail processing empty `$styles` and ensure we don't leverage cache in development mode.

I'm also adding an expiry time for the cache keys — a day is a reasonable value. This will [help the system to clean up old data](https://github.com/memcached/memcached/wiki/UserInternals#when-memory-is-reclaimed) that it's not used often, in case there's a need for resources.